services/libs/tinybird/datasources/segmentRepositories.datasource

```diff
@@ -0,0 +1,26 @@
+DESCRIPTION >
+    - `segmentRepositories` contains the repository data associated with segments.
+    - Replicated from Postgres - it is meant to become the single source of truth about repositories in the future.
+    - Schema:
+    - `repository` is the URL for the repository.
+    - `segmentId` links to the segment the repository belongs to.
+    - `insightsProjectId` links to the insightsProject the repository belongs to.
+    - `createdAt` is a standard timestamp field to record lifecycle tracking.
+    - `archived` indicates whether the repository is archived (true/false).
+    - `excluded` indicates whether the repository is excluded from analytics and metrics (true/false).
+    - `last_archived_check` is the timestamp of the last check for whether the repository is archived.
+
+SCHEMA >
+    `repository` String `json:$.record.repository`,
+    `segmentId` UUID `json:$.record.segmentId`,
+    `insightsProjectId` UUID `json:$.record.insightsProjectId`,
+    `createdAt` DateTime64(3) `json:$.record.createdAt`,
+    `archived` Bool `json:$.record.archived`,
+    `excluded` Bool `json:$.record.excluded`,
+    `last_archived_check` Nullable(DateTime64(3)) `json:$.record.last_archived_check`
+
+
+ENGINE ReplacingMergeTree
+ENGINE_PARTITION_KEY toYear(createdAt)
+ENGINE_SORTING_KEY repository
+ENGINE_VER createdAt
```

Comment:
You'll need an `updatedAt` field to keep track of different versions of the same row. `createdAt` won't change on consecutive updates to the same row (when it's archived or excluded, for instance). So it can even pick the old row when an update happens

Reply:
Oh, yeah, I forgot about that. The original table Umberto created doesn't have it and I wondered if it was necessary, but forgot to check it.

Reply:
You also need to change the `ENGINE_VER` to `updatedAt` so that this field is used for versioning